.github/ISSUE_TEMPLATE/bug.yml

```diff
@@ -0,0 +1,55 @@
+name: Bug Report
+description: Report an issue or bug with the Quantum Core X.
+title: "bug: "
+labels: [ "type/bug" ]
+body:
+  - type: markdown
+    attributes:
+      value: |
+        Please fill out the sections below to help us diagnose and fix the issue. Before submitting a bug report, please check the [Reporting Bugs](https://github.com/MeikelLP/quantum-core-x/blob/master/CONTRIBUTING.md#reporting-bugs) section in the contributing guidelines.
+
+  - type: textarea
+    id: what-happened
+    attributes:
+      label: What happened?
+      description: A clear and concise description of what actually happened.
+      placeholder: Describe the bug you encountered.
+    validations:
+      required: true
+
+  - type: textarea
+    id: how-to-reproduce
+    attributes:
+      label: How to reproduce?
+      description: Steps to reproduce the behavior.
+      placeholder: Steps to reproduce the behavior.
+    validations:
+      required: true
+
+  - type: textarea
+    id: tracebacks
+    attributes:
+      label: Tracebacks
+      description: If applicable, paste the full traceback here.
+      placeholder: Paste the full traceback here.
+
+  - type: dropdown
+    id: os-platform
+    attributes:
+      label: OS Platform
+      description: Which OS platform are you using?
+      options:
+        - Windows
+        - macOS
+        - Linux
+        - Other (please specify)
+    validations:
+      required: true
```

Comment:
Should we replace the OS Platform drop-down with a more generic OS & platform section, with pre-filled example template along the lines of:

```md
# Windows: `winver`
# Linux: `cat /etc/os-release`
# Mac: `sw_vers`
- OS Name: Windows / Linux / Mac / etc.
- OS Distribution: Windows Home / Pro / Server || Debian / Ubuntu || etc.
- OS Version: 10 / 11 24H2 || 12.7 / 24.04.1 || etc.

- CPU: 12 Cores @ 4.2 GHz
- RAM: 32 GB

- Optional:
  - Machine Type: bare-metal / virtual machine
```

@MeikelLP How much system and platform information is useful for troubleshooting? I'd figure you could use as much information as possible, without making the issue template too much of a chore, while still avoiding immediately having to ask follow-up questions like which Linux distribution and version / kernel it is?

Do we need the output of `dotnet --list-runtimes` ?

Reply:
I can't estimate how much this information is worth. Currently it's worth not much - may be more relevant in the future. I'd again keep it simple for now

Reply:
In that case I will mark this as resolved, we can just add it later if needed.